src/data-source/collection-browser-data-source.ts

```diff
@@ -246,6 +246,19 @@ export class CollectionBrowserDataSource
     this.requestHostUpdate();
   }
 
+  /**
+   * @inheritdoc
+   */
+  resetPages(): void {
+    if (Object.keys(this.pages).length !== this.host.maxPagesToManage) {
+      this.pages = {};
```

Comment:
Q: Do we need to reset `this.numTileModels` too since we are clearing them?

Reply:
We can but reset this create blank result placeholder while fetching 750 records. so just reset pages makes the  placeholder types to say, large search in-progress